### PR TITLE
[SPARK-24303][PYTHON] Update cloudpickle to v0.4.4

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

cloudpickle 0.4.4 is released - https://github.com/cloudpipe/cloudpickle/releases/tag/v0.4.4

There's no invasive change - the main difference is that we are now able to pickle the root logger, which fix is pretty isolated.

## How was this patch tested?

Jenkins tests.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @ueshin and @BryanCutler

File: python/pyspark/cloudpickle.py [hunk 1]
```diff
@@ -801,10 +801,10 @@ def save_ellipsis(self, obj):
     def save_not_implemented(self, obj):
         self.save_reduce(_gen_not_implemented, ())
 
-    if PY3:
-        dispatch[io.TextIOWrapper] = save_file
-    else:
+    try:               # Python 2
         dispatch[file] = save_file
+    except NameError:  # Python 3
+        dispatch[io.TextIOWrapper] = save_file
```
Comment (felixcheung) [inline python/pyspark/cloudpickle.py]: is it our preference to check for python version or handle error?

Comment (HyukjinKwon) [reply python/pyspark/cloudpickle.py]: Actually not quite sure if we should do the same thing in our PySpark codebase .. but it's the decision made in cloudpickle .. it's virtually the same anyway

**Review by BryanCutler — APPROVED** [review]
LGTM, thanks @HyukjinKwon !

**Comment by ueshin** [comment]
LGTM.

**Comment by BryanCutler** [comment]
merged to master

**Pull request closed without merging**